### PR TITLE
fix: Error catch logic for when status poll update fail on one environment update 

### PR DESCRIPTION
**Pull request by Bingjiling**
Issue #, if available:

Description of changes:

* Error catch logic for when status poll update fail on one environment update 
* Add console log before the first DB scan is issued to track time 
* Corresponding Unit test 
* Use existing api handler role for status poll Lambda as a temporary patch

Checklist: 

* [x] Have you successfully deployed to an AWS account with your changes? 
* [x] Have you linted your code locally prior to submission?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran unit tests and manual tests with your changes locally?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by SanketD92 — DISMISSED** [review]
LGTM

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js [hunk 1]
```diff
@@ -94,6 +94,7 @@ class EnvironmentScService extends Service {
 
   async pollAndSyncWsStatus(requestContext) {
     const [indexesService, awsAccountsService] = await this.service(['indexesService', 'awsAccountsService']);
+    console.log('Start DB scan for status poll and sync.');
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js:97]: You don't want to use console.log.  You want to use this.log.info() instead.

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js [hunk 2]
```diff
@@ -143,17 +144,25 @@ class EnvironmentScService extends Service {
...
+          // Log the error and skip the update for now
+          // The next invocation of poll and sync will do the sync if it's still needed
+          await this.update(requestContext, newEnvironment);
+          ec2Updated[ec2InstanceId] = {
+            ddbID: existingEnvRecord.id,
+            currentStatus: expectedDDBStatus,
+            staleStatus: existingEnvRecord.status,
+          };
+        } catch (e) {
+          console.log(`Error updating record ${existingEnvRecord.id}`);
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js:163]: Please see my comments about consoe.log()

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js [hunk 3]
```diff
@@ -204,7 +213,15 @@ class EnvironmentScService extends Service {
...
           status: SageMakerStatusMap[sagemakerRealtimeStatus[key]].toUpperCase(),
         };
-        await this.update(requestContext, newEnvironment);
+        try {
+          // Might run into situation where the environment was just updated and rev number does not match
+          // Log the error and skip the update for now
+          // The next invocation of poll and sync will do the sync if it's still needed
+          await this.update(requestContext, newEnvironment);
+        } catch (e) {
+          console.log(`Error updating record ${existingEnvRecord.id}`);
```
Comment (hatimkhan) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js:222]: Do you need to propagate  this exception? if so, then you don't want to log it here and instead use the boom conventions for throwing errors.

**Review by SanketD92 — DISMISSED** [review]
LGTM

**Review by SanketD92 — APPROVED** [review]
LGTM

**Pull request merged**